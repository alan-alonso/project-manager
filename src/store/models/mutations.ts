import { ProjectStatus } from "@/models/project";

export interface ChangeProjectStatusPayload {
  projectId: string;
  newStatus: ProjectStatus;
}
