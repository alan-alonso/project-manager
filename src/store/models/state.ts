import { Project } from "@/models/project";

export interface StateModel {
  projects: Project[];
}
