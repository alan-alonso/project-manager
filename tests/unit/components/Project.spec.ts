import Vue from "vue";
import Vuetify from "vuetify";

Vue.use(Vuetify);

import { createLocalVue, shallowMount } from "@vue/test-utils";
import ProjectComp from "@/components/Project.vue";
import { Project, ProjectStatus } from "@/models/project";

const localVue = createLocalVue();

describe("Project.vue", () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let vuetify: any;
  let project: Project;

  beforeEach(() => {
    vuetify = new Vuetify();
    project = new Project(
      "1",
      "Test Title",
      "Test Description",
      1,
      ProjectStatus.ACTIVE
    );
  });

  test("should render component with project prop data", () => {
    // act
    const wrapper = shallowMount(ProjectComp, {
      localVue,
      vuetify,
      propsData: {
        project
      }
    });

    // assert
    expect(wrapper.text()).toContain(project.title);
    expect(wrapper.text()).toContain(project.description);
    expect(wrapper.text()).toContain(project.people);
  });

  test("renders project item correctly", () => {
    // act
    const wrapper = shallowMount(ProjectComp, {
      localVue,
      vuetify,
      propsData: {
        project
      }
    });

    // assert
    expect(wrapper.element).toMatchSnapshot();
  });
});
