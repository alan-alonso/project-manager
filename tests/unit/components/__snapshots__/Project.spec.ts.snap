// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project.vue renders project item correctly 1`] = `
<v-card-stub
  draggable=""
  loaderheight="4"
  tag="div"
>
  <v-card-title-stub>
    Test Title
  </v-card-title-stub>
   
  <v-card-subtitle-stub>
    <v-icon-stub>
      people
    </v-icon-stub>
    
    1
  
  </v-card-subtitle-stub>
   
  <v-card-text-stub>
    <p>
      Test Description
    </p>
  </v-card-text-stub>
</v-card-stub>
`;
